integration-test/src/test/java/org/apache/iotdb/db/it/builtinfunction/scalar/IoTDBSubStrFunctionIT.java

```diff
@@ -0,0 +1,175 @@
+/*
```

Comment:
Check if the tests cover the following cases for both new/old transformer:

legal:
1. only specify start position
2. specify start and end position

illegal:
1. using substring without start and end position.
3. using substring with float start position like substring(s1,1.0)
4. negative start position 
5. start position >= end position
6. end position > length
7. wrong input type